lib/queueservice.js

```diff
@@ -0,0 +1,82 @@
+import xml2js from 'xml2js';
+import Debug from 'debug';
+import request from 'superagent-promise';
+
+let debug = Debug('taskcluster-docker-worker:queueService');
+
+export default class TaskQueue {
+  constructor(config) {
+    this.queues;
+    this.workerType = config.workerType;
+    this.provisionerId = config.provisionerId;
+    this.client = config.queue;
+    this.log = config.log;
+  }
+
+  async getQueues() {
+    if (!this.queues || (new Date(this.queues.expires)).getTime() > Date.now()) {
```

Comment:
How about giving yourself 5 min to pull the queue... Ie.
`new Date(this.queues.expires)).getTime() > Date.now() + 5 * 60 * 1000`

Ideally we should validate this before every request. But that non-trivial the queue may return a different number of queues next time... Anyways, 5 min should enough to poll the entire list.